lib/index.js

```diff
@@ -479,6 +488,86 @@ Socket.prototype.send = function(msg, flags) {
   return this;
 };
 
+/**
+ * Read one message from socket.
+ *
+ * @return {Array} multi-part message or null if socket doesn't have any
+ * @api public
+ */
+Socket.prototype.read = function() {
+  var message = [], flags;
+
+  if (this._zmq.state !== zmq.STATE_READY) {
+    return null;
+  }
+
+  flags = this._zmq.getsockopt(zmq.ZMQ_EVENTS);
+
+  if (flags & zmq.ZMQ_POLLIN) {
+    do {
+      message.push(this._zmq.recv());
+    } while (this._zmq.getsockopt(zmq.ZMQ_RCVMORE));
+
+    return message;
+  }
+
+  return null;
+}
+
+/**
+ * Send the given `msg`.
+ *
+ * @param {String|Buffer|Array} msg
+ * @param {Function} callback function (optional)
+ * @return {Socket} for chaining
+ * @api public
+ */
+Socket.prototype.write = function (parts, callback) {
+  var self = this,
+    i = 0,
+    len = parts.length,
+    flags = this._zmq.getsockopt(zmq.ZMQ_EVENTS),
+    part;
+
+  if (!Array.isArray(parts)) {
+    parts = [parts];
+  }
+
+  // socket is not ready to accept messages
+  if (!(flags & zmq.ZMQ_POLLOUT)) {
+    server.once('ready', function () {
+      self.write(parts, callback);
+    });
+    return;
+  }
+
+  while (i < len) {
+    try {
+      part = parts[i];
+      if (!Buffer.isBuffer(part)) {
+        part = new Buffer(String(part), 'utf8');
+      }
+      if (i === len - 1) {
+        this._zmq.send(part, 0);
+      } else {
+        this._zmq.send(part, zmq.ZMQ_SNDMORE);
+      }
+    } catch (sendError) {
+      // More chunks were to follow, which we should now drop.
+      // This loop will pull off the items up until and including
+      // the first item that is not flagged SNDMORE.
```

Comment:
not needed anymore, write accepts String|Buffer|Array and doesn't accept flag param, so we can just drop the current multipart message